### PR TITLE
CoreDataStack

### PR DESCRIPTION
**Pull request by AnastasiaOsinska**

### CODE REVIEW

File: ToDo+CoreDataProperties.swift [hunk 1]
```diff
@@ -0,0 +1,26 @@
+//
+//  ToDo+CoreDataProperties.swift
```
Comment (vadimux) [inline ToDo+CoreDataProperties.swift:2]: we should remove this file

File: ToDo+CoreDataClass.swift [hunk 2]
```diff
@@ -0,0 +1,15 @@
+//
+//  ToDo+CoreDataClass.swift
```
Comment (vadimux) [inline ToDo+CoreDataClass.swift:2]: we should remove this file and select codeGenerationType="class" for CoreData Entity

File: TodoApp/DataProvider/CoreDataManager.swift [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    }
+    
+    func deleteAllData(entity: String) {
+        let context = persistentContainer.viewContext
+        let fetchRequest = NSFetchRequest<NSFetchRequestResult>(entityName: entity)
+        fetchRequest.returnsObjectsAsFaults = false
+        do {
+            let results = try context.fetch(fetchRequest)
+                for managedObject in results {
+                    let managedObjectData:NSManagedObject = managedObject as! NSManagedObject
```
Comment (vadimux) [inline TodoApp/DataProvider/CoreDataManager.swift:68]: I don't recommend use force unwrapping for this

File: TodoApp/DataProvider/CoreDataManager.swift [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+            print(toDo)
+            saveContext()
+        }
+    }
+    
+    //MARK: NSFetchedResultsControllerDelegate
+    
+    func controller(_ controller: NSFetchedResultsController<NSFetchRequestResult>, didChange anObject: Any, at indexPath: IndexPath?, for type: NSFetchedResultsChangeType, newIndexPath: IndexPath?) {
+        
+        switch type {
```
Comment (vadimux) [inline TodoApp/DataProvider/CoreDataManager.swift:90]: is this `switch` really needed here?

File: Resources/AppDelegate.swift [hunk 5]
```diff
@@ -6,10 +6,11 @@
 //
 
 import UIKit
+import CoreData
 
 @main
 class AppDelegate: UIResponder, UIApplicationDelegate {
-
```
Comment (vadimux) [inline Resources/AppDelegate.swift:12]: such changes should not be committed

File: TodoApp/DataProvider/CoreDataManager.swift [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+class CoreDataManager: NSObject, NSFetchedResultsControllerDelegate {
+    
+    static let shared = CoreDataManager()
+    private var fetchedRC: NSFetchedResultsController<ToDo>!
+    
+    private override init() { }
+    
+    lazy var persistentContainer: NSPersistentContainer = {
+        let container = NSPersistentContainer(name: "ToDo")
+        container.loadPersistentStores(completionHandler: {
```
Comment (vadimux) [inline TodoApp/DataProvider/CoreDataManager.swift:20]: if the `completionHandler` has the last position - we can omit it

File: TodoApp/DataProvider/CoreDataManager.swift [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+                try context.save()
+            } catch {
+                let err = error as NSError
+                fatalError("Unresolved error \(err), \(err.userInfo)")
+            }
+        }
+    }
+   
+    func readFromCoreData(completion: @escaping(_ result: Result<[Todo], Error>) -> Void) {
+        var allTableDataArray : [ToDo] = [ToDo]()
```
Comment (vadimux) [inline TodoApp/DataProvider/CoreDataManager.swift:44]: extra space before colon

File: TodoApp/DataProvider/CoreDataManager.swift [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+        let context = persistentContainer.viewContext
+        let fetchRequest = NSFetchRequest<ToDo>(entityName: "ToDo")
+        do {
+            allTableDataArray = try context.fetch(fetchRequest)
+            var newTodos = [Todo]()
```
Comment (vadimux) [inline TodoApp/DataProvider/CoreDataManager.swift:49]: the same names for classes are very inconvenient, maybe we should think about a different naming?

**Review by vadimux — CHANGES_REQUESTED** [review]

File: ToDo+CoreDataClass.swift (hunk 2 above)
Comment (AnastasiaOsinska) [reply ToDo+CoreDataClass.swift:2]: done

File: TodoApp/DataProvider/CoreDataManager.swift (hunk 3 above)
Comment (AnastasiaOsinska) [reply TodoApp/DataProvider/CoreDataManager.swift:68]: done

**Pull request merged**